### PR TITLE
Allow custom jinja environments

### PR DESCRIPTION
**Pull request by ThiefMaster**
This is useful e.g. when using the new Jinja Environment
attributes added in mitsuhiko/jinja2#404

### CODE REVIEW

File: flask/app.py [hunk 1]
```diff
@@ -157,6 +157,11 @@ class Flask(_PackageBoundObject):
     #: :class:`~flask.Response` for more information.
     response_class = Response
 
+    #: The class that is used for the Jinja environment.
+    #:
+    #: .. versionadded:: 0.11
```
Comment (ThiefMaster) [inline flask/app.py]: Not sure if this should be 0.11 or 1.0 - currently both versions are used, with 1.0 being more common but the current package version in `flask/__init__.py` being 0.11

Comment (untitaker) [inline flask/app.py]: I always assumed it is 1.0 but I'm not sure whether we should release an intermediate version.

Comment (ThiefMaster) [inline flask/app.py]: Hm I guess for now the different versions don't really matter as long as it's cleaned up before whatever the next release's version number will be ;)

**Comment by untitaker** [comment]
Thanks @ThiefMaster!

**Pull request merged**